### PR TITLE
#99: First stage of maven 3 migration

### LINKED ISSUE #143: Introduce JSR330
I think this plugin could benefit from dependency injection.

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Also resolves #143: migration to JSR330

This is the first stage of Maven 3 migration: metadata retrieval and archetype retrieval left for the next stages.

@slawekjaranowski please review

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@slachiewicz please review

**Comment by slawekjaranowski** [comment]
Thank, I will try review in next week 😄

**Comment by ajarmoniuk** [comment]
> Thank, I will try review in next week smile

Kindly reminder :smile:

File: pom.xml [hunk 1]
```diff
@@ -120,7 +120,7 @@
       <dependency>
         <groupId>org.apache.maven</groupId>
         <artifactId>maven-artifact-manager</artifactId>
-        <version>${mavenVersion}</version>
+        <version>2.2.1</version>
```
Comment (slawekjaranowski) [inline pom.xml:123]: I assume, that will be removed in next step

File: pom.xml [hunk 2]
```diff
@@ -148,9 +148,9 @@
         <version>4.6.1</version>
       </dependency>
       <dependency>
-        <groupId>org.apache.maven</groupId>
-        <artifactId>maven-project</artifactId>
-        <version>${mavenVersion}</version>
+        <groupId>org.apache.maven.resolver</groupId>
+        <artifactId>maven-resolver-api</artifactId>
```
Comment (slawekjaranowski) [inline pom.xml:152]: Should be `org.eclipse.aether:aether-api` to be consistent with Maven 3.2.5

**Pull request merged**